### PR TITLE
OLS-201: Check HEAD HTTP method for OpenAPI endpoint

### PR DESCRIPTION
**Pull request by tisnik**
## Description

Check HEAD HTTP method for OpenAPI endpoint

## Type of change

- [ ] Refactor
- [ ] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up dependent library
- [x] Integration tests

## Related Tickets & Documents

- Related Issue #[OLS-201](https://issues.redhat.com//browse/OLS-201)

### CODE REVIEW

File: tests/integration/test_openapi.py [hunk 1]
```diff
@@ -42,3 +42,10 @@ def test_openapi_endpoint():
     paths = payload["paths"]
     for endpoint in ("/readiness", "/liveness", "/v1/query", "/v1/feedback"):
         assert endpoint in paths, f"Endpoint {endpoint} is not described"
+
+
+def test_openapi_endpoint_head_method():
+    """Check if REST API allows HEAD HTTP method for endpoint with OpenAPI specification."""
+    response = client.head("/openapi.json")
```
Comment (onmete) [inline tests/integration/test_openapi.py:49]: We are not implementing it, right? This comes from FastAPI itself. Does it make sense to test it though?

Comment (tisnik) [reply tests/integration/test_openapi.py:49]: I think yes, as these are integration tests. So it check  the integration with FastAPI and configuration there.

Comment (onmete) [reply tests/integration/test_openapi.py:49]: I can't say I understand the benefit of such a test, but ok. :)

**Pull request merged**